src/scripts/main.js

```diff
@@ -1,3 +1,33 @@
 'use strict';
 
-// write code here
+// variables
+const list = document.querySelector('ul');
+const employees = [...list.children];
+
+// create a list of employees from the data on the page
+function getEmployees(employeesList) {
+  return employeesList.map((employee) => ({
+    name: employee.innerText,
+    position: employee.dataset.position,
+    salary: employee.dataset.salary,
+    age: +employee.dataset.age,
+  }));
+}
+
+getEmployees(employees);
+
+// function for removing chars: $ and ','
+function removeChars(str) {
+  return +str.replaceAll(',', '').replaceAll('$', '');
+}
+
+// sort the list by salary in descending order
+const sortedEmployeesList = employees.sort((empA, empB) => {
+  const salaryA = removeChars(empA.dataset.salary);
+  const salaryB = removeChars(empB.dataset.salary);
+
+  return salaryB - salaryA;
+});
```

Comment:
If I'm not mistaken, this part should be the second function.
But who knows, maybe it's really not necessary, anyway your solution is great.